WordPress/Classes/ViewRelated/Blog/My Site/MySiteViewController.swift

```diff
@@ -623,7 +654,7 @@ class MySiteViewController: UIViewController, NoResultsViewHost {
             return
         }
 
-        remove(blogDashboardViewController)
+        removeChildFromStackView(blogDashboardViewController)
```

Comment:
By doing this, you're forcing this VC to be deleted/regenerated every time the user switches between them.

This in turn, causes the collection view on the dashboard to always recreate itself (including cells, child VCs, etc) — the same for the blog details (it is recreated every time, and it will perform the `preloadMetadata` every time, which is a heavy API call). Is that done on purpose? If so, why?

Reply:
Actually, BlogDetailsVC / BlogDashboardVC isn't recreated every time we switch. `viewDidLoad` is triggered only once for each child vc. You can add a breakpoint in viewDidLoad / deinit to verify this.`viewDidAppear`, however, is called every time we switch — which is expected behavior.

The only difference between `remove(blogDashboardViewController)` and `removeChildFromStackView(blogDashboardViewController)` is that the latter explicitly removes the child VC view from the stackview's arranged subviews, but this should have no effect on the lifecycle of the child VCs.